### ISSUE TITLE
Fail to get the access token

### ISSUE THREAD

**Issue by ming060** [ngxson/psid-to-fbid#1]
First of all, thank you for this great module.

I followed the instructions of getting started guide, in the page source, I can't' find the access token start with `EAAA` or anything like that. Did I miss something or Facebook already fixed this and not revealing this token in the page source?

Thank you for your help in advance.

**Comment by ngxson** [comment]
Thanks for notifying me. After the security issue that leaks access_token through "View as" function, facebook hided the access_token from profile's source code.

However, you can still use the restserver API, which is extracted from facebook's iOS app:

```
function genLoginURLiOS(email, pass) {
    return "https://api.facebook.com/restserver.php?api_key=3e7c78e35a76a9299309885393b02d97&email="+encodeURIComponent(email)+"&format=JSON&generate_session_cookies=1&locale=vi_vn&method=auth.login&password="+encodeURIComponent(pass)+"&return_ssl_resources=0&v=1.0&sig=" + 
    md5("api_key=3e7c78e35a76a9299309885393b02d97email="+email+"format=JSONgenerate_session_cookies=1locale=vi_vnmethod=auth.loginpassword="+pass+"return_ssl_resources=0v=1.0c1e620fa708a1d5696fb991c1bde5662")
}
```

This function will give you an URL. Open this link in the browser and you will receive the access_token. It will be implemented in the next version.

**Comment by ming060** [comment]
Hi @ngxson ,

Thanks for getting back to me so quickly.
I follow this way and get my access token successfully.

But when I use this token and my page id to initiate the module, I encountered below error and I can't init the module correctly. The code I used is just copy & paste from the get started guide and replace the page_id and token with my own ones.

Do you have any idea on how to solve this?

Thanks again.

    /app/node_modules/psid-to-fbid/index.js:43

					data.data.forEach(page => {

					          ^


    TypeError: Cannot read property 'forEach' of undefined

    at Request._callback (/app/node_modules/psid-to-fbid/index.js:43:16)

    at Request.self.callback (/app/node_modules/request/request.js:185:22)

    at emitTwo (events.js:126:13)

    at Request.emit (events.js:214:7)

    at Request.<anonymous> (/app/node_modules/request/request.js:1161:10)

    at emitOne (events.js:116:13)

    at Request.emit (events.js:211:7)

    at IncomingMessage.<anonymous> (/app/node_modules/request/request.js:1083:12)

    at Object.onceWrapper (events.js:313:30)

    at emitNone (events.js:111:20)

**Comment by ming060** [comment]
Hi @ngxson ,

It's me again.

I try to fix this by just skipping the checking.

    if (page.id == page_id) {
        self.state.page_token = page.access_token;
    }

And I can now get the fbid successfully.
I'm pretty new to node and I am not sure why the checking failed.

I thought the checking should be like this:

    if (page.id ==  this.state.page_id) {
        self.state.page_token = page.access_token;
    }

But it still failed.

Maybe you can give me some direction on fixing this.

Thanks.

**Comment by ngxson** [comment]
Sorry I've made a mistake at `page.id == page_id`, so the right code should be `page.id == self.state.page_id`

The `fetchPageToken` function just get the page's access_token from your personal access_token, by doing GET request to https://graph.facebook.com/me/accounts. Normally I supplied page's access_token directly via
```new PsidToFbid(page_id, {page_token: page_token})```
so I haven't seen this problem.

Thank you for notifying me. Feel free to reply to this thread if you still have problems.

**Comment by ming060** [comment]
Thank you for the support.

I'm now facing another problem and unable to get the fbid successfully on a published and reviewed messenger bot.
I'm not sure where the problem is. I saw some error messages like GET not support. I hope it's not blocked by the Facebook side.

I will post another issue and provide more information.

**Comment by ming060** [comment]
> Sorry I've made a mistake at `page.id == page_id`, so the right code should be `page.id == self.state.page_id`
> 
> The `fetchPageToken` function just get the page's access_token from your personal access_token, by doing GET request to https://graph.facebook.com/me/accounts. Normally I supplied page's access_token directly via
> `new PsidToFbid(page_id, {page_token: page_token})`
> so I haven't seen this problem.
> 
> Thank you for notifying me. Feel free to reply to this thread if you still have problems.

So if I use a page_token to init, can I still get the fbid?

**Comment by ngxson** [comment]
Of course, the module will work just fine. However, for example, you have dozens of pages that use this module, and some day all tokens are expired, then you will have to refresh them manually.

The `fetchPageToken` function do it automatically, so you can simply supply 1 personal access_token, then the code will obtain all page tokens.
